Implement Var and Final Where Applicable.

klortiz13 opened the pull request:
## Summary
Following up with our effort to improve the Campus App's Codebase (we've done the Try/Catch/Finally and a Network Helper Library), we are now looking for using Var and final wherever it is applicable to avoid data type repetition and significantly condense the codebase (improving readability).
I also added more comments, fixed typos, kept a consistent format, one lined a lot of conditional statements and condensed one-line functions.

## Changelog
[General] [Change] - Cleaned up the codebase to improve readability and maintain consistency.


## Test Plan
Ensure no behavior was impacted by these changes. Everything should work as if nothing was changed.

klortiz13 committed:
concised providers

klortiz13 committed:
finished services and all other files but ui

klortiz13 committed:
finished ui

klortiz13 committed:
finished second runthrough of providers

klortiz13 committed:
finished second runthrough of services

klortiz13 committed:
finished core

klortiz13 committed:
finished ui

morebytes commented on lib/core/providers/cards.dart:
```diff
@@ -22,13 +22,11 @@ class CardsDataProvider extends ChangeNotifier {
-  Map<String, bool> _cardStates = {};
-  Map<String, CardsModel?> _webCards = {};
-
+  bool? _isLoading = false;
```
This change makes _isLoading nullable, which is generally a regression unless absolutely required.

klortiz13 committed:
fixed regression

klortiz13 committed:
merged with experimental

klortiz13 committed:
fixing provider conflicts

klortiz13 committed:
fixed providers, still missing free food

klortiz13 committed:
Maas

a
i:erge branch 'experimental' of https://github.com/klortiz13/my-campus-mobile-2 into var-final-work
